actix-http/src/config.rs

```diff
@@ -230,72 +232,132 @@ impl fmt::Write for Date {
     }
 }
 
-#[derive(Clone)]
-struct DateService(Rc<DateServiceInner>);
+/// Service for update Date and Instant periodically at 500 millis interval.
+struct DateService {
+    current: Rc<Cell<(Date, Instant)>>,
+    handle: JoinHandle<()>,
+}
 
-struct DateServiceInner {
-    current: Cell<Option<(Date, Instant)>>,
+impl Drop for DateService {
+    fn drop(&mut self) {
+        // stop the timer update async task on drop.
+        self.handle.abort();
+    }
 }
 
-impl DateServiceInner {
+impl DateService {
     fn new() -> Self {
-        DateServiceInner {
-            current: Cell::new(None),
-        }
+        // shared date and timer for DateService and update async task.
+        let current = Rc::new(Cell::new((Date::new(), Instant::now())));
+        let current_clone = current.clone();
+        // spawn an async task sleep for 500 milli and update current date/timer in a loop.
+        // handle is used to stop the task on DateService drop.
+        let handle = actix_rt::spawn(async move {
+            #[cfg(test)]
+            let _notify = notify_on_drop::NotifyOnDrop::new();
+
+            let mut interval = interval(Duration::from_millis(500));
+            loop {
+                // TODO: consider use the Instant returned by interval stream
+                // (If it's accurate enough)
+                let _now = interval.tick().await;
```

Comment:
it does not appear there would be any lag to using this returned value

Reply:
Could be happen when server is blocked by user code for a long time or there are too much polling on other futures like FuturesUnordered or JoinAll and delay the interval. The interval return the Instant from the timer's deadline in timer wheel rather than generate a new Instant.

I'm just guessing and have not test this.

Reply:
should not occur with join all or futures unordered
that said, if user is blocking thread or has written a poll busy loop they got bigger problems than an outdated Date header; could even be a useful signal that something is wrong